### PR TITLE
Update rich to >=9 and update [] escaping

### PR DESCRIPTION
**Pull request by ewels**
Minor bump to Rich, as a change to how it does escaping meant that all of the square brackets in the logs were showing up as double square brackets.

## PR checklist

 - [x] This comment contains a description of changes (with reason)
 - [x] `CHANGELOG.md` is updated
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] Documentation in `docs` is updated

### CODE REVIEW

File: nf_core/lint.py [hunk 1]
```diff
@@ -1346,11 +1347,11 @@ def _s(some_list):
         table = Table(box=rich.box.ROUNDED)
         table.add_column("[bold green]LINT RESULTS SUMMARY".format(len(self.passed)), no_wrap=True)
```
Comment (ggabernet) [inline nf_core/lint.py:1348]:
```suggestion
        table.add_column(r"\[bold green]LINT RESULTS SUMMARY".format(len(self.passed)), no_wrap=True)
```

Comment (ggabernet) [reply nf_core/lint.py:1348]: Not sure if it should also be escaped here?

**Review by ggabernet — APPROVED** [review]
Looks good to me!

File: nf_core/lint.py (hunk 1 above)
Comment (ewels) [reply nf_core/lint.py:1348]: Sure - this didn't trigger a warning but can't hurt I guess 👍🏻

**Comment by ewels** [comment]
ah crap, the `r""` thing breaks the rendering, hang on

**Pull request merged**